README.md
# setiadi3_walanae


changes.txt
We took the SLiMS 7 source to build Setiadi 2
====================================
Setiadi Info
====================================
Developers :
Dwi Fajar Saputra (dfsptra@gmail.com) -> Inisiator & Programmer
Drajat Hasan (drajathasan20@gmail.com) -> Programmer
Erwan Setyo Budi (erwans818@gmail.com) -> Web Design

Status of Setiadi Release :
Ready  : The current ready release.
Testing : The next generation release.
InBuild : The preparation release.

==============================================
initialization of Setiadi CodeName :

Frozen : The developer decide yet the codename.

==============================================

==============================================
Release Status Current : 

"InBuild" with codename "Walanae"

========================================================================
Setiadi 3 Changelog
========================================================================

++++++++++++++++++++
Update in 31-08-2020
++++++++++++++++++++
Added	: Update Template Setiadi2 Walanae
Added	: Add Student ID
Added : Add Contact Email
Added : Add Supervisor
Added : Add Examiner
Added : Add Contributor
Added : Add Code Ministry / Kode Prodi PDDIKTI
Fixed : Update Date Type
Added : Add Copyright
Added : Add License
Added : Add URL Croosref
Added : Add Send Notification Email Approved
Added : Add Masterfile Supervisor
Added : Add Masterfile Examiner
Added : Add Masterfile Code Ministry
Added : Add Masterfile license
Update : Detail Template Biblio
Update : Sidebar Right 
Update : Sidebar Left





==============================================

==============================================
Release Status Current : 

"InBuild" with codename "Lukulo"

========================================================================
Setiadi 2 Changelog
========================================================================
++++++++++++++++++++
Update in 01-04-2018
++++++++++++++++++++
Added	: Font Awesome 5
Added	: Content in dashboard

++++++++++++++++++++
Update in 31-03-2018
++++++++++++++++++++
Fixed	: Undisable calendar button in biblio edit mode
Fixed	: Open calendar calendar top position
Added	: Style for input type button
Fixed	: Error in memberlist

++++++++++++++++++++
Update in 30-03-2018 20:37:40
++++++++++++++++++++
Fixed	: Template 'Null' value when update system configuration
Added	: Write Log function in simple-theme.php

++++++++++++++++++++
Update in 30-03-2018
++++++++++++++++++++
Fixed : OAI PMH function
Fixed : UCS node function
Fixed : Nayanes node function

++++++++++++++++++++
Update in 29-03-2018
++++++++++++++++++++
Updated	: Codename from 'Frozen' to 'Lukulo'

++++++++++++++++++++
Update in 28-03-2018
++++++++++++++++++++
Updated : Cookies name and Changes from SLiMS 7  (Cendana) to Setiadi 2 (Frozen)
Updated : Password encryption method with password_hash() 
Fixed	: SQL Interupted error information
Updated	: Main admin template function and Ui component
Updated : Setiadi Form Maker Class
Added	: Setiadi Utitility Class
Added	: Setiadi Shorcut wizard
Added	: Setiadi Theme Configuration

++++++++++++++++++++
Update in 26-03-2018 21:51:24 
++++++++++++++++++++
Updated : Color and Icon in setboot_function.inc.php
Updated : Module system and change bibliography to ETD
Added	: Simple-theme.php and simple-them.inc.php for changing theme with preview image and theme information

++++++++++++++++++++
Update in 26-03-2018  
++++++++++++++++++++
Added	: Subject field
Added	: Function Select without custom field, "+" button in Setiadi Form Maker
Updated	: submenu.php in module reporting and master_file

++++++++++++++++++++
Update in 25-03-2018  23:18:03 
++++++++++++++++++++
Fixed : getTreeView.php for error

++++++++++++++++++++
Update in 25-03-2018  22:07:37 
++++++++++++++++++++
Added	: getColor function in file setboot_function.inc.php
Added    : bootstrap css in  file printed_page_tpl.php 
Updated	: Change icon on navbar from glyphicon to themify icon
Updated	: file index_template.inc.php add #chartDiv css 
Fixed	: Visualize Diagram Type Collection

++++++++++++++++++++
Update in 25-03-2018  
++++++++++++++++++++
Updated : Bibliography submenu (deleted itemOut)
Updated	: hide form component in PopUp mode

++++++++++++++++++++
Update in 24-03-2018  19:54:45 
++++++++++++++++++++
Updated : Setiadi Form Maker Class
Fixed	: msg information and Ui component in PopUP mode Biblio form

++++++++++++++++++++
Update in 24-03-2018  
++++++++++++++++++++
Updated : Setiadi Form Maker Class
Added : blocker link click in biblio cover image edit mode
Added : Image upload form component
Added : Image Cover in Biblio Edit mode

++++++++++++++++++++
Update in 23-03-2018 13:35:04 
++++++++++++++++++++
Updated : Setiadi Form Maker
Fixed      : Select filter

++++++++++++++++++++
Update in 23-03-2018 
++++++++++++++++++++
Fixed : Ui componen
Added: table long_pattern
Added: Setiadi Utitlity Function
Added: Attachment
Added: Author
Added: Item Batch

+++++++++++++++++++++
Update in 22-03-2018 
+++++++++++++++++++++
Added: Edit Mode For Setiadi Form Maker Class
Fixed  : Biblio Step
Added: Ui Component
Added: Select Function in simbiodbop

+++++++++++++++++++++

+++++++++++++++++++++
current update
add new column in biblio
'ALTER TABLE slims.biblio ADD step varchar(100) DEFAULT 1 NOT NULL ;'

bibliography/index.php
form_maker/simbio_form_maker.inc.php

current update 02-11-2017
[mod] 
template/default/css/style.css
template/default/index_template.inc.php
lib/detail.inc.php
lib/biblio_list_model.inc.php

[added]
magnifier to directory js

[mod]
template/default/index_template.inc.php

current update 3/9/2017
====
[added]
driftjs to directory js

[mod]
template/default/index_template.inc.php
lib/detail.inc.php
template/default/detail_template.php

current update (28-08-2017)
===============================================
[added] 
--treantjs to directory "js"
--getTreeView.php in directory "admin/modules/reporting/"
--treeview.php in directory "admin/modules/reporting"

[modified]

admin/modules/system/content.php
<?php

/**
 *
 * Copyright (C) 2007,2008  Arie Nugraha (dicarve@yahoo.com)
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 *
 */

/* Dynamic content Management section */

// key to authenticate
define('INDEX_AUTH', '1');
// key to get full database access
define('DB_ACCESS', 'fa');

// main system configuration
require '../../../sysconfig.inc.php';
// IP based access limitation
require LIB . 'ip_based_access.inc.php';
do_checkIP('smc');
do_checkIP('smc-system');
// start the session
require SB . 'admin/default/session.inc.php';
require SB . 'admin/default/session_check.inc.php';
require SIMBIO . 'simbio_GUI/form_maker/simbio_form_table_AJAX.inc.php';
require SIMBIO . 'simbio_GUI/template_parser/simbio_template_parser.inc.php';
require SIMBIO . 'simbio_GUI/table/simbio_table.inc.php';
require SIMBIO . 'simbio_GUI/paging/simbio_paging.inc.php';
require SIMBIO . 'simbio_DB/datagrid/simbio_dbgrid.inc.php';
require SIMBIO . 'simbio_DB/simbio_dbop.inc.php';

// privileges checking
$can_read = utility::havePrivilege('system', 'r');
$can_write = utility::havePrivilege('system', 'w');

if (!$can_read) {
    die('<div class="errorBox">' . __('You don\'t have enough privileges to view this section') . '</div>');
}

/* RECORD OPERATION */
if (isset($_POST['saveData'])) {
    $contentTitle = trim(strip_tags($_POST['contentTitle']));
    $contentPath = trim(strip_tags($_POST['contentPath']));
    // check form validity
    if (empty($contentTitle) or empty($contentPath)) {
        utility::jsAlert(__('Title or Path can\'t be empty!'));
        exit();
    } else {
        $data['content_title'] = $dbs->escape_string(strip_tags(trim($contentTitle)));
        $data['content_path'] = strtolower($dbs->escape_string(strip_tags(trim($contentPath))));
        $data['content_desc'] = $dbs->escape_string(trim($_POST['contentDesc']));
        $data['input_date'] = date('Y-m-d H:i:s');
        $data['last_update'] = date('Y-m-d H:i:s');

        // create sql op object
        $sql_op = new simbio_dbop($dbs);
        if (isset($_POST['updateRecordID'])) {
            /* UPDATE RECORD MODE */
            // remove input date
            unset($data['input_date']);
            // filter update record ID
            $updateRecordID = (int) $_POST['updateRecordID'];
            // update the data
            $update = $sql_op->update('content', $data, 'content_id=' . $updateRecordID);
            if ($update) {
                // write log
                utility::writeLogs($dbs, 'staff', $_SESSION['uid'], 'system', $_SESSION['content_title'] . ' update content data (' . $data['content_title'] . ') with contentname (' . $data['contentname'] . ')');
                utility::jsAlert(__('Content data updated'));
                echo '<script type="text/javascript">parent.$(\'#mainContent\').simbioAJAX(parent.$.ajaxHistory[0].url);</script>';
            } else {
                utility::jsAlert(__('Content data FAILED to update!') . "\nDEBUG : " . $sql_op->error);
            }
            exit();
        } else {
            /* INSERT RECORD MODE */
            // insert the data
            if ($sql_op->insert('content', $data)) {
                // write log
                utility::writeLogs($dbs, 'staff', $_SESSION['uid'], 'system', $_SESSION['realname'] . ' add new content (' . $data['content_title'] . ') with contentname (' . $data['contentname'] . ')');
                utility::jsAlert(__('Content data saved'));
                echo '<script type="text/javascript">parent.$(\'#mainContent\').simbioAJAX(\'' . $_SERVER['PHP_SELF'] . '\');</script>';
            } else {
                utility::jsAlert(__('Content data FAILED to save!') . "\n" . $sql_op->error);
            }
            exit();
        }
    }
    exit();
} else if (isset($_POST['itemID']) and !empty($_POST['itemID']) and isset($_POST['itemAction'])) {
    if (!($can_read and $can_write)) {
        die();
    }
    /* DATA DELETION PROCESS */
    $sql_op = new simbio_dbop($dbs);
    $failed_array = array();
    $error_num = 0;
    if (!is_array($_POST['itemID'])) {
        // make an array
        $_POST['itemID'] = array((int) $_POST['itemID']);
    }
    // loop array
    foreach ($_POST['itemID'] as $itemID) {
        $itemID = (int) $itemID;
        // get content data
        $content_q = $dbs->query('SELECT content_title FROM content WHERE content_id=' . $itemID);
        $content_d = $content_q->fetch_row();
        if (!$sql_op->delete('content', "content_id='$itemID'")) {
            $error_num++;
        } else {
            // write log
            utility::writeLogs($dbs, 'staff', $_SESSION['uid'], 'system', $_SESSION['realname'] . ' DELETE content (' . $content_d[0] . ')');
        }
    }

    // error alerting
    if ($error_num == 0) {
        utility::jsAlert(__('All Data Successfully Deleted'));
        echo '<script type="text/javascript">parent.$(\'#mainContent\').simbioAJAX(\'' . $_SERVER['PHP_SELF'] . '?' . $_POST['lastQueryStr'] . '\');</script>';
    } else {
        utility::jsAlert(__('Some or All Data NOT deleted successfully!\nPlease contact system administrator'));
        echo '<script type="text/javascript">parent.$(\'#mainContent\').simbioAJAX(\'' . $_SERVER['PHP_SELF'] . '?' . $_POST['lastQueryStr'] . '\');</script>';
    }
    exit();
}
/* RECORD OPERATION END */

/* search form */
?>
<fieldset class="menuBox">
    <div class="menuBoxInner systemIcon">
        <div class="per_title">
            <h2><?php echo __('Content'); ?></h2>
        </div>
        <div class="sub_section">
            <div class="btn-group">
                <a href="<?php echo MWB; ?>system/content.php" class="btn btn-default"><i class="glyphicon glyphicon-list-alt"></i>&nbsp;<?php echo __('Content List'); ?></a>
                <a href="<?php echo MWB; ?>system/content.php?action=detail" class="btn btn-default"><i class="glyphicon glyphicon-plus"></i>&nbsp;<?php echo __('Add New Content'); ?></a>
            </div>
            <form name="search" action="<?php echo MWB; ?>system/content.php" id="search" method="get" style="display: inline;"><?php echo __('Search'); ?> :
                <input type="text" name="keywords" size="30" />
                <input type="submit" id="doSearch" value="<?php echo __('Search'); ?>" class="btn btn-default" />
            </form>
        </div>
    </div>
</fieldset>
<?php
/* main content */
if (isset($_POST['detail']) or (isset($_GET['action']) and $_GET['action'] == 'detail')) {
    if (!($can_read and $can_write)) {
        die('<div class="errorBox">' . __('You don\'t have enough privileges to view this section') . '</div>');
    }
    /* RECORD FORM */
    // try query
    $itemID = (int) isset($_POST['itemID']) ? $_POST['itemID'] : 0;
    $rec_q = $dbs->query('SELECT * FROM content WHERE content_id=' . $itemID);
    $rec_d = $rec_q->fetch_assoc();

    // create new instance
    $form = new simbio_form_table_AJAX('mainForm', $_SERVER['PHP_SELF'] . '?' . $_SERVER['QUERY_STRING'], 'post');
    $form->submit_button_attr = 'name="saveData" value="' . __('Save') . '" class="btn btn-default"';

    // form table attributes
    $form->table_attr = 'align="center" id="dataList" cellpadding="5" cellspacing="0"';
    $form->table_header_attr = 'class="alterCell" style="font-weight: bold;"';
    $form->table_content_attr = 'class="alterCell2"';

    // edit mode flag set
    if ($rec_q->num_rows > 0) {
        $form->edit_mode = true;
        // record ID for delete process
        // form record id
        $form->record_id = $itemID;
        // form record title
        $form->record_title = $rec_d['content_title'];
        // submit button attribute
        $form->submit_button_attr = 'name="saveData" value="' . __('Update') . '" class="btn btn-default"';
    }

    /* Form Element(s) */
    // content title
    $form->addTextField('text', 'contentTitle', __('Content Title') . '*', $rec_d['content_title'], 'style="width: 100%;"');
    // content path
    $form->addTextField('text', 'contentPath', __('Path (Must be unique)') . '*', $rec_d['content_path'], 'style="width: 50%;"');
    // content description
    $form->addTextField('textarea', 'contentDesc', __('Content Description'), htmlentities($rec_d['content_desc'], ENT_QUOTES), 'style="width: 100%; height: 500px;"');

    // edit mode messagge
    if ($form->edit_mode) {
        echo '<div class="infoBox">' . __('You are going to update Content data'), ' : <b>' . $rec_d['content_title'] . '</b> <br />' . __('Last Updated') . $rec_d['last_update'] . '</div>'; //mfc
    }
    // print out the form object
    // init TinyMCE instance
    echo '<script type="text/javascript">tinyMCE.init({
        // Options
        mode : "exact", elements : "contentDesc", theme : "advanced",
        plugins : "safari,style,table,media,searchreplace,directionality,visualchars,xhtmlxtras,compat2x",
        skin : "o2k7", skin_variant : "silver",
        // Theme options
        theme_advanced_buttons1 : "bold,italic,underline,strikethrough,|,justifyleft,justifycenter,justifyright,justifyfull,styleselect,fontsizeselect,|,bullist,numlist,|,link,unlink,anchor,image,cleanup,code,forecolor",
        theme_advanced_buttons2 : "undo,redo,|,tablecontrols,|,hr,removeformat,visualaid,|,charmap,media,|,ltr,rtl,|,search,replace",
        theme_advanced_buttons3 : null, theme_advanced_toolbar_location : "top", theme_advanced_toolbar_align : "center", theme_advanced_statusbar_location : "bottom",
        theme_advanced_resizing : false, content_css : "' . (SWB . 'admin/' . $sysconf['admin_template']['css']) . '",
        });</script>';
    echo $form->printOut();
} else {
    /* USER LIST */
    // table spec
    $table_spec = 'content AS c';

    // create datagrid
    $datagrid = new simbio_datagrid();
    if ($can_read and $can_write) {
        $datagrid->setSQLColumn(
            'c.content_id',
            'c.content_title AS \'' . __('Content Title') . '\'',
            'c.content_path AS \'' . __('Path (Must be unique)') . '\'',
            'c.last_update AS \'' . __('Last Updated') . '\''
        );
    } else {
        $datagrid->setSQLColumn(
            'c.content_title AS \'' . __('Content Title') . '\'',
            'c.content_path AS \'' . __('Path (Must be unique)') . '\'',
            'c.last_update AS \'' . __('Last Updated') . '\''
        );
    }
    $datagrid->setSQLorder('c.last_update DESC');

    // is there any search
    $criteria = 'c.content_id IS NOT NULL ';
    if (isset($_GET['keywords']) and $_GET['keywords']) {
        $keywords = $dbs->escape_string($_GET['keywords']);
        $criteria .= " AND MATCH(content_title, content_desc) AGAINST('$keywords')";
    }
    $datagrid->setSQLCriteria($criteria);

    // set table and table header attributes
    $datagrid->table_attr = 'align="center" id="dataList" cellpadding="5" cellspacing="0"';
    $datagrid->table_header_attr = 'class="dataListHeader" style="font-weight: bold;"';
    // set delete proccess URL
    $datagrid->chbox_form_URL = $_SERVER['PHP_SELF'];

    // put the result into variables
    $datagrid_result = $datagrid->createDataGrid($dbs, $table_spec, 20, ($can_read and $can_write));
    if (isset($_GET['keywords']) and $_GET['keywords']) {
        $msg = str_replace('{result->num_rows}', $datagrid->num_rows, __('Found <strong>{result->num_rows}</strong> from your keywords')); //mfc
        echo '<div class="infoBox">' . $msg . ' : "' . $_GET['keywords'] . '"</div>';
    }

    echo $datagrid_result;
}
/* main content end */
